dev_scripts/boxplot.py

```diff
@@ -0,0 +1,188 @@
+import random
+import utils.math as m
+import matplotlib.pyplot as plt
+from matplotlib.patches import Patch
+from reporting.generate import get_sorted_scores
+from utils.ui import colour_print
+
+
+CUSTOM_COLORS = ["#ff9999", "#66b3ff", "#99ff99", "#ffcc99", "#76d7c4", "#ffb3e6", "#c4e17f", "#c2c2f0"]
+
+# Agent names may vary over time, so link them here with their corresponding short labels.
+# The order is important, and will be kept the same in the figure / table.
+# The script expects results from all aliases listed here, so remove any agent from both
+# the aliases and context_lengths dicts if there are no results for them.
+aliases = {
+    "LLMChatSession - together_ai-mistralai-Mixtral-8x7B-Instruct-v0.1 - 32000": "Mixtral 8x7B",
+    "LLMChatSession - together_ai-mistralai-Mixtral-8x7B-Instruct-v0.1 - 32768": "Mixtral 8x7B",
+    "LLMChatSession - together_ai-mistralai-Mixtral-8x22B-Instruct-v0.1 - 65536": "Mixtral 8x22B",
+    "LLMChatSession - together_ai-meta-llama-Llama-3-70b-chat-hf - 8000": "Llama 3 70B",
+    "LLMChatSession - gpt-3.5-turbo - 16384": "GPT-3.5 turbo",
+    "LLMChatSession - gpt-4-turbo - 128000": "GPT-4 turbo",
+    "LLMChatSession - gpt-4-turbo-2024-04-09 - 128000": "GPT-4 turbo",
+    "LLMChatSession - gpt-4o - 128000": "GPT-4o",
+    "LLMChatSession - gpt-4o-2024-05-13 - 128000": "GPT-4o",
+    "LLMChatSession - claude-3-opus - 200000": "Claude 3 Opus",
+    "GeminiProInterface": "Gemini 1.5 Pro",
+    "LTMAgentWrapper - claude-3-opus-20240229 - 16384 - QG_JSON_USER_INFO": "LTM Claude 3 Opus",
+    "LTMAgentWrapper - claude-3-opus - 16384 - QG_JSON_USER_INFO": "LTM Claude 3 Opus",
+    "LTMAgentWrapper - gpt-4-turbo-2024-04-09 - 16384 - QG_JSON_USER_INFO": "LTM GPT-4 turbo",
+    "LTMAgentWrapper - gpt-4-turbo - 16384 - QG_JSON_USER_INFO": "LTM GPT-4 turbo",
+    "LTMAgentWrapper - together_ai-meta-llama-Llama-3-70b-chat-hf - 8000 - QG_JSON_USER_INFO": "LTM Llama 3 70B",
+}
```

Comment:
Minor point: Could add in the memgpt results to the bottom here

Reply:
I'd rather merge it as it is now and change this dict locally, since we(you)'ll probably be adding and updating results. Finally, you could just make a PR with your local version. Does it sound ok?

Reply:
yeah, fine